cli/src/pcluster/models/imagebuilder.py

```diff
@@ -0,0 +1,166 @@
+        components: List[Component] = None,
+    ):
+        super().__init__()
+        self.name = name
+        self.description = description
+        self.tags = tags
+        self.parameter_store_name_output = parameter_store_name_output
+        self.root_volume = root_volume
+        self.components = components
+        self._default_value()
+        # TODO: add validator
+
+    def _default_value(self):
+        if not self.tags:
+            self.tags = []
+        self.tags.append(MarkedTag("PclusterVersion", "".join(utils.get_installed_version())))
+
+
+# ---------------------- Build ---------------------- #
+
+
+class Build(Resource):
+    """Represent the build configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        instance_type: str,
+        parent_image: str,
+        instance_role: str = None,  # TODO: auto generate if not assigned
+        subnet_id: str = None,  # TODO: auto generate if not assigned
+        tags: List[Tag] = None,
+        security_group_ids: List[str] = None,
+    ):
+        super().__init__()
+        self.instance_type = instance_type
+        self.parent_image = parent_image
+        self.instance_role = instance_role
+        self.tags = tags
+        self.subnet_id = subnet_id
+        self.security_group_ids = security_group_ids
+        # TODO: add validator
+
+
+# ---------------------- Dev Settings ---------------------- #
+
+
+class ChefCookbook(Resource):
+    """Represent the chef cookbook configuration for the ImageBuilder."""
+
+    def __init__(self, url: str, json: str):
+        super().__init__()
+        self.url = url
+        self.json = json
+        # TODO: add validator
+
+
+class DevSettings(Resource):
+    """Represent the dev settings configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        update_os_and_reboot: bool = None,
+        disable_pcluster_component: bool = None,
+        chef_cookbook: ChefCookbook = None,
+        node_url: str = None,
+        aws_batch_cli_url: str = None,
+        distribution_configuration_arn: str = None,
+        terminate_instance_on_failure: bool = None,
+    ):
+        super().__init__()
+        self.update_os_and_reboot = update_os_and_reboot
+        self.disable_pcluster_component = disable_pcluster_component
+        self.chef_cookbook = chef_cookbook
+        self.node_url = node_url
+        self.aws_batch_cli_url = aws_batch_cli_url
+        self.distribution_configuration_arn = distribution_configuration_arn
+        self.terminate_instance_on_failure = terminate_instance_on_failure
+        # TODO: add validator
+
+    def _default_value(self):
+        if not self.update_os_and_reboot:
+            self.update_os_and_reboot = MarkedBool(False)
+        if not self.disable_pcluster_component:
+            self.disable_pcluster_component = MarkedBool(False)
+        if not self.terminate_instance_on_failure:
+            self.terminate_instance_on_failure = MarkedBool(True)
```

Comment:
Is it a bug? What if `self.terminate_instance_on_failure` is False? I assume in your code, the attribute will always be `True`?

Reply:
yes, it is a bug, not `self.terminate_instance_on_failure` should be `self.terminate_instance_on_failure` is `None`

Reply:
Changed